test/e2e/operator_test.go

```diff
@@ -670,3 +672,112 @@ func TestHostNetworkEndpointPublishingStrategy(t *testing.T) {
 		t.Fatalf("WARNING: cloud resources may have been leaked! failed to delete ingresscontroller %s: %v", name, err)
 	}
 }
+
+// TestInternalLoadBalancer creates an ingresscontroller with the
+// "LoadBalancerService" endpoint publishing strategy type with scope set to
+// "Internal" and verifies that the operator creates a load balancer and that
+// the load balancer has a private IP address.
+func TestInternalLoadBalancer(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	infraConfig := &configv1.Infrastructure{}
+	err = cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig)
+	if err != nil {
+		t.Fatalf("failed to get infrastructure config: %v", err)
+	}
+
+	switch infraConfig.Status.Platform {
+	case configv1.AWSPlatformType:
+	case configv1.AzurePlatformType:
+	case configv1.BareMetalPlatformType:
+		t.Skip("test skipped on bare metal")
+	case configv1.GCPPlatformType:
+	case configv1.LibvirtPlatformType:
+		t.Skip("test skipped on libvirt")
+	case configv1.OpenStackPlatformType:
+	case configv1.NonePlatformType:
+		t.Skip("test skipped on \"none\" platform type")
+	case configv1.VSpherePlatformType:
+		t.Skip("test skipped on vSphere")
+	}
+
+	dnsConfig := &configv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
+		t.Fatalf("failed to get DNS 'cluster': %v", err)
+	}
+
+	name := types.NamespacedName{Namespace: ns, Name: ingressControllerName}
+	domain := name.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newIngressController(name.Name, name.Namespace, domain, operatorv1.LoadBalancerServiceStrategyType)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = &operatorv1.LoadBalancerStrategy{
+		Scope: operatorv1.InternalLoadBalancer,
+	}
+	if err := cl.Create(context.TODO(), ic); err != nil {
```

Comment:
Move the delete to a defer after this create and you can reduce the branching/nesting level of subsequent checks

Reply:
You mean by changing `if  ... { t.Errorf(foo) } else { t.Errorf(bar) }` to `if ... { t.Fatalf(foo) } t.Errorf(bar)`?

Reply:
Yeah, exactly